Bottom: ed4802ba7856d42e86e862034bd30044a93010e5
Top:    6a58b0c24f53edc0308b2d390e08e2d7838db90c
Author: Parth Shah <parth@linux.ibm.com>
Date:   2020-01-09 14:18:45 +0530

Refresh of sched-core-update-turbo_sched

---

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index b156eacbb0b1..9fa91d72793f 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -5023,7 +5023,8 @@ static int __sched_setscheduler(struct task_struct *p,
 	prev_class = p->sched_class;
 
 	/* Refcount tasks classified as a small background task */
-	(p->latency_tolerance > 18) ? turbo_sched_get() : turbo_sched_put();
+	if (p->latency_tolerance > 18 != attr->latency_tolerance > 18)
+		(p->latency_tolerance > 18) ? turbo_sched_get() : turbo_sched_put();
 
 	__setscheduler(rq, p, attr, pi);
 	__setscheduler_uclamp(p, attr);
